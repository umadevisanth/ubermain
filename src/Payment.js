import React, { useState } from 'react';
import { View, Text, SafeAreaView, TouchableOpacity, StyleSheet, Image, Modal } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Globalstyle from './Globalstyle';

const Payment = () => {
  const [selectedPayment, setSelectedPayment] = useState(null);  // State to track selected payment method
  const [showAlert, setShowAlert] = useState(false);  // State to control the display of the alert
  const [showModal, setShowModal] = useState(false);  // State to control the display of the modal
  const navigation = useNavigation();  // Hook to get navigation object

  // Payment method labels and messages
  const payment = {
    later: 'DO THIS LATER',
    payment: 'Select your preferred payment method',
    credit: 'Credit or Debit card',
    paytm: 'Paytm',
    cash: 'Cash',
    next: 'Next',
    ride: 'You wonâ€™t be able to request a ride without adding a payment method.',
    pay: 'Add Payment Method Now',
    later: 'Do It Later',
    confirm: 'Confirm Payment',
  };

  // Handler for payment method selection
  const handlePress = (method) => {
    setSelectedPayment(method);
    setShowModal(true);
  };

  // Handler for 'Do This Later' button
  const handleLaterPress = () => {
    setShowAlert(true);  // Show the alert message
  };

  // Handler to add a payment method
  const handleAddPaymentMethod = () => {
    setShowAlert(false);  // Close the alert
    navigation.navigate('Payment');  // Navigate to Payment screen
  };

  // Handler to close the alert and navigate to Pickup screen
  const handleCloseAlert = () => {
    setShowAlert(false);  // Close the alert
    navigation.navigate('Paylater');  // Navigate to Pickup screen
  };

  // Handler for confirming the payment method
  const handleConfirmPayment = () => {
    setShowModal(false);
    navigation.navigate('Success');  // Navigate to the Success screen
  };

  return (
    <SafeAreaView style={Globalstyle.paymentsafeareaview}>
      <View style={Globalstyle.paymentviewtwxt}>
        {/* 'Do This Later' button */}
        <TouchableOpacity onPress={handleLaterPress}>
          <Text style={Globalstyle.Payment_later}>{payment.later}</Text>
        </TouchableOpacity>

        {/* Payment method selection header */}
        <View style={Globalstyle.paymentselection}>
          <Text style={Globalstyle.Payment_payment}>{payment.payment}</Text>
        </View>

        {/* Payment method options */}
        <View style={Globalstyle.paymentmethodoptions}>
          <TouchableOpacity onPress={() => handlePress(payment.credit)}>
            <View style={Globalstyle.Payment_creditcard}>
              <Image style={Globalstyle.paymentmethodimage} source={require('../assets/CreditCard.png')} />
              <Text style={Globalstyle.Payment_debitcard}>{payment.credit}</Text>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => handlePress(payment.paytm)}>
            <View style={Globalstyle.Payment_paytm}>
              <Image style={Globalstyle.paymentmethodimage} source={require('../assets/Paytm.png')} />
              <Text style={Globalstyle.Paytm_text}>{payment.paytm}</Text>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => handlePress(payment.cash)}>
            <View style={Globalstyle.Paytm_opacity}>
              <Image style={Globalstyle.paymentmethodimage} source={require('../assets/Cash.png')} />
              <Text style={Globalstyle.Paytm_cash}>{payment.cash}</Text>
            </View>
          </TouchableOpacity>
        </View>

        {/* Alert message */}
        {showAlert && (
          <View style={Globalstyle.alertContainer}>
            <Text style={Globalstyle.alertMessage}>{payment.ride}</Text>
            <TouchableOpacity style={[Globalstyle.button, Globalstyle.addButton]} onPress={handleAddPaymentMethod}>
              <Text style={[Globalstyle.buttonText, Globalstyle.addButtonText]}>{payment.pay}</Text>
            </TouchableOpacity>
            <TouchableOpacity style={[Globalstyle.button, Globalstyle.laterButton]} onPress={handleCloseAlert}>
              <Text style={[Globalstyle.buttonText, Globalstyle.laterButtonText]}>{payment.later}</Text>
            </TouchableOpacity>
          </View>
        )}

        {/* 'Next' button to proceed with the selected payment method */}
        <View style={Globalstyle.paymentbuttontext}>
          {selectedPayment && (
            <TouchableOpacity style={Globalstyle.Payment_next} onPress={handleConfirmPayment}>
              <Text style={Globalstyle.paymentnexttext}>{payment.next}</Text>
            </TouchableOpacity>
          )}
        </View>
      </View>

      {/* Modal for confirming payment method */}
      <Modal visible={showModal} transparent animationType="slide">
        <View style={Globalstyle.modalContainer}>
          <View style={Globalstyle.modalContent}>
            <Text style={Globalstyle.modalText}>Confirm Payment via {selectedPayment}?</Text>
            <TouchableOpacity style={Globalstyle.confirmButton} onPress={handleConfirmPayment}>
              <Text style={Globalstyle.confirmButtonText}>{payment.confirm}</Text>
            </TouchableOpacity>
            <TouchableOpacity style={Globalstyle.cancelButton} onPress={() => setShowModal(false)}>
              <Text style={Globalstyle.cancelButtonText}>Cancel</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({

});

export default Payment;
