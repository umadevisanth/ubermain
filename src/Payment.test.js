import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Payment from '../src/Payment';

// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  AntDesign: 'AntDesign', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('Login component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Payment />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
