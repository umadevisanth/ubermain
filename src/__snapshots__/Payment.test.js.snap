// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <AntDesign
        color="black"
        name="left"
        size={20}
        style={
          {
            "color": "#EDF6FF",
            "height": 20,
            "left": 40,
            "top": 40,
            "width": 20,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#535AFF",
          "fontSize": 20,
          "left": 230,
          "lineHeight": 23.44,
          "top": 80,
        }
      }
    >
      DO THIS LATER
    </Text>
  </View>
  <View>
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 19,
          "height": 33,
          "left": 38,
          "top": 131,
          "width": 341,
        }
      }
    >
      Select your preferred payment method
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "left": 40,
          "top": 200,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/credit_card_PNG28.png",
          }
        }
        style={
          {
            "height": 50,
            "width": 50,
          }
        }
      />
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "left": 50,
            "top": 15,
          }
        }
      >
        Credit or Debit card
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "left": 30,
          "padding": 20,
          "top": 200,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/free.jpg",
          }
        }
        style={
          {
            "height": 50,
            "width": 50,
          }
        }
      />
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "left": 50,
            "top": 15,
          }
        }
      >
        Paytm
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "left": 30,
          "top": 200,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/money.jpg",
          }
        }
        style={
          {
            "height": 50,
            "width": 50,
          }
        }
      />
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "left": 50,
            "top": 15,
          }
        }
      >
        Cash
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#EDF6FF",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 58,
        "justifyContent": "center",
        "left": 37,
        "opacity": 1,
        "top": 400,
        "width": 341,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        {
          "color": "white",
          "fontSize": 21,
          "left": 150,
        }
      }
    >
      Next
    </Text>
  </View>
</RCTSafeAreaView>
`;
